### PR TITLE
add more descriptorVersion args

### PR DESCRIPTION
**Pull request by gluhar2006**

### CODE REVIEW

File: lunavl/sdk/estimators/face_estimators/warper.py [hunk 1]
```diff
@@ -26,22 +23,27 @@ class WarpedImage(VLImage):
...
+        if isinstance(body, VLImage):
+            self.source = body.source
+            self.filename = body.filename
+            self.coreImage = body.coreImage
         else:
-            self.source = vlImage.source
-            self.filename = vlImage.filename
-            self.coreImage = vlImage.coreImage
+            super().__init__(body, filename=filename, colorFormat=colorFormat)
+            self.assertWarp()
```
Comment (matemax) [inline lunavl/sdk/estimators/face_estimators/warper.py]:
```suggestion
        self.assertWarp()
```

File: lunavl/sdk/faceengine/descriptors.py [hunk 2]
```diff
@@ -200,18 +202,22 @@ def generateDescriptor(
             else:
                 faceDescriptor.reload(descriptor=descriptor)
         else:
-            faceDescriptor = FaceDescriptor(self._faceEngine.coreFaceEngine.createDescriptor())
+            faceDescriptor = FaceDescriptor(self._faceEngine.coreFaceEngine.createDescriptor(self._descriptorVersion))
```
Comment (matemax) [inline lunavl/sdk/faceengine/descriptors.py]: + descriptorVersion to args

File: lunavl/sdk/faceengine/descriptors.py [hunk 3]
```diff
@@ -167,10 +167,12 @@ class FaceDescriptorFactory:
 
     Attributes:
         _faceEngine (VLFaceEngine): faceEngine
+        _descriptorVersion (int): descriptor version
```
Comment (matemax) [inline lunavl/sdk/faceengine/descriptors.py]: + description for 0

File: lunavl/sdk/faceengine/descriptors.py [hunk 4]
```diff
@@ -200,18 +202,22 @@ def generateDescriptor(
...
     @CoreExceptionWrap(LunaVLError.CreationDescriptorError)
-    def generateDescriptorsBatch(self, size: int) -> FaceDescriptorBatch:
+    def generateDescriptorsBatch(self, size: int, descriptorVersion: int = 0) -> FaceDescriptorBatch:
         """
         Generate core descriptors batch.
 
         Args:
-            size:batch size
+            size: batch size
+            descriptorVersion: descriptor version
```
Comment (matemax) [inline lunavl/sdk/faceengine/descriptors.py]: + description for 0

File: lunavl/sdk/faceengine/engine.py [hunk 5]
```diff
@@ -172,7 +172,7 @@ def createAGSEstimator(self) -> AGSEstimator:
         """
         return AGSEstimator(self._faceEngine.createAGSEstimator())
 
-    def createFaceDescriptorEstimator(self, descriptorVersion: Optional[int] = None) -> FaceDescriptorEstimator:
+    def createFaceDescriptorEstimator(self, descriptorVersion: int = 0) -> FaceDescriptorEstimator:
```
Comment (matemax) [inline lunavl/sdk/faceengine/engine.py:175]: + description for 0

**Review by matemax — APPROVED** [review]

File: lunavl/sdk/faceengine/descriptors.py (hunk 2 above)
Comment (gluhar2006) [reply lunavl/sdk/faceengine/descriptors.py]: ok

File: lunavl/sdk/faceengine/descriptors.py (hunk 3 above)
Comment (gluhar2006) [reply lunavl/sdk/faceengine/descriptors.py]: ok

File: lunavl/sdk/faceengine/engine.py (hunk 5 above)
Comment (gluhar2006) [reply lunavl/sdk/faceengine/engine.py:175]: ok

File: lunavl/sdk/faceengine/descriptors.py [hunk 6]
```diff
@@ -200,18 +203,23 @@ def generateDescriptor(
             else:
                 faceDescriptor.reload(descriptor=descriptor)
         else:
-            faceDescriptor = FaceDescriptor(self._faceEngine.coreFaceEngine.createDescriptor())
+            faceDescriptor = FaceDescriptor(
+                self._faceEngine.coreFaceEngine.createDescriptor(descriptorVersion or self._descriptorVersion))
```
Comment (matemax) [inline lunavl/sdk/faceengine/descriptors.py]: make property for valid access to `_descriptorVersion` not from `FaceDescriptorFactory`

```python
@property
def descriptorVersion(self) -> int:
   """
   Return default  descriptor version for generating descriptor.
   Returns:
       _descriptorVersion
   '""
    return self._descriptorVersion
```

**Review by matemax — APPROVED** [review]

File: lunavl/sdk/faceengine/descriptors.py (hunk 6 above)
Comment (gluhar2006) [reply lunavl/sdk/faceengine/descriptors.py]: ok

**Pull request merged**